include/sys/atomic.h

```diff
@@ -32,7 +32,10 @@ typedef void *atomic_ptr_t;
 # ifdef CONFIG_XTENSA
 /* Not all Xtensa toolchains support GCC-style atomic intrinsics */
 # include <arch/xtensa/atomic_xtensa.h>
-# endif
+# else
+/* Other arch specific implementation */
+# include <sys/atomic_arch.h>
+# endif /* CONFIG_XTENSA */
```

Comment:
Should it be more explicit to convey out-of-tree architecture and have documentation in porting/architectures supported sections?

Reply:
As this is not explicitly targeting out-of-tree architecture but could be anyone using a compiler without support for the built-in (as mentioned in the documentation for [ATOMIC_OPERATIONS_ARCH](ATOMIC_OPERATIONS_ARCH), I'm a little in doubt about mentioning that.
Let me know what you think.
But I do agree that, since the built-in method is default, a note of selecting the ATOMIC_OPERATIONS_ARCH if using this should be added to the the porting guides [Utility Libraries](https://docs.zephyrproject.org/latest/guides/porting/arch.html#utility-libraries) section .
I will add that.